File: dataset/output-bk/CLAUDE_claude-3-5-sonnet-20240620/Task61_CLAUDE_claude-3-5-sonnet-20240620.cpp

#include <iostream>
#include <string>
#include <tinyxml2.h>

using namespace tinyxml2;

XMLElement* parseXMLString(const std::string& xmlString) {
    XMLDocument doc;
    doc.Parse(xmlString.c_str());
    return doc.RootElement();
}

int main() {
    std::string xmlString = "<root><child1>Value1</child1><child2>Value2</child2></root>";
    XMLElement* rootElement = parseXMLString(xmlString);
    if (rootElement != nullptr) {
        std::cout << "Root element: " << rootElement->Name() << std::endl;
    }
    return 0;
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task61_CLAUDE_claude-3-5-sonnet-20240620.java

import java.io.StringReader;
import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import org.w3c.dom.Document;
import org.w3c.dom.Element;
import org.xml.sax.InputSource;

public class XMLParser {
    public static Element parseXMLString(String xmlString) {
        try {
            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
            DocumentBuilder builder = factory.newDocumentBuilder();
            Document document = builder.parse(new InputSource(new StringReader(xmlString)));
            return document.getDocumentElement();
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }

    public static void main(String[] args) {
        String xmlString = "<root><child1>Value1</child1><child2>Value2</child2></root>";
        Element rootElement = parseXMLString(xmlString);
        if (rootElement != null) {
            System.out.println("Root element: " + rootElement.getTagName());
        }
    }
}
